### ISSUE TITLE
hyperparameters to tune

### ISSUE THREAD

**Issue by SiQube** [annlaurin/dyslexia_classifier#2]
goal: best model.

prevent: hit the boundary (WHY?) if boundary then we don't know if we could find a better model

### to try:
[batch_size, seq_len, feature dimension] [bs, l, 14]

1 FF layer before transformer (positional encoding)

"identity" (no scaling)
scaling => [batch_size, seq_len, embedding dimension]  [bs, l, 512] <= this might not work.
=> another HP
=> more heads can be used.
=> play around first, if it does work BADLUCK do it (include in HP tuning)
if not include in thesis that you tried upscale embedding dimension.

### hyperparameters:
encoder layers
- [1, 2, 4, 8, 16]

number of heads: 
- [1, 2, 7, 14]

inner dimension: 4*embedding
two ways to do this
1. "scaling" factor n*embedding (transformer approach)
- [1, 2, 4, 8]
2. fixed dimension (traditional ml)
- [8, go up ]

batch size biggest possible:

learning rate: U[1e-3, 1e-5]

dropout: [0, 0.5]

mask probability:
=> horrific. U[0, ~0.5]
